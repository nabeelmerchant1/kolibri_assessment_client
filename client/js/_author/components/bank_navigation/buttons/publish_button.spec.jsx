import React          from 'react';
import { shallow }    from 'enzyme';
import PublishButton  from './publish_button';

describe('PublishButton component', () => {
  let props;
  let result;
  let focused;
  let toggling;

  beforeEach(() => {
    focused = false;
    toggling = false;
    props = {
      togglePublishAssessment: () => { toggling = true; },
      onFocus: () => { focused = true; },
      assessment: {
        isToggling: true,
      },
    };

    result = shallow(<PublishButton {...props} />);
  });

  it('matches a snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('calls onFocus', () => {
    expect(focused).toBeFalsy();
    result.find('button').simulate('focus');
    expect(focused).toBeTruthy();
  });

  it('calls togglePublishAssessment on click', () => {
    expect(toggling).toBeFalsy();
    props.assessment.isToggling = false;
    result = shallow(<PublishButton {...props} />);
    result.find('button').simulate('click', { stopPropagation: () => {} });
    expect(toggling).toBeTruthy();
  });
});
