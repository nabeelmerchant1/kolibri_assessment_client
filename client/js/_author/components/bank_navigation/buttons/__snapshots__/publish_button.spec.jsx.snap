// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishButton component matches a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="au-c-btn au-c-btn--square au-c-publish "
    onClick={[Function]}
    onFocus={[Function]}
>
    <i
        className="material-icons"
    >
        cloud_upload
    </i>
</button>,
  "nodes": Array [
    <button
      className="au-c-btn au-c-btn--square au-c-publish "
      onClick={[Function]}
      onFocus={[Function]}
>
      <i
            className="material-icons"
      >
            cloud_upload
      </i>
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <PublishButton
        assessment={
                Object {
                        "isToggling": true,
                      }
        }
        onFocus={[Function]}
        togglePublishAssessment={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "assessment": Object {
            "isToggling": true,
          },
          "onFocus": [Function],
          "togglePublishAssessment": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="au-c-btn au-c-btn--square au-c-publish "
          onClick={[Function]}
          onFocus={[Function]}
>
          <i
                    className="material-icons"
          >
                    cloud_upload
          </i>
</button>,
        "_debugID": 2,
        "_renderedOutput": <button
          className="au-c-btn au-c-btn--square au-c-publish "
          onClick={[Function]}
          onFocus={[Function]}
>
          <i
                    className="material-icons"
          >
                    cloud_upload
          </i>
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PublishButton
    assessment={
        Object {
            "isToggling": true,
          }
    }
    onFocus={[Function]}
    togglePublishAssessment={[Function]}
/>,
}
`;
